flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/CreditBasedSequenceNumberingViewReader.java

```diff
@@ -163,8 +172,15 @@ boolean hasBuffersAvailable() {
 		return subpartitionView.isAvailable(Integer.MAX_VALUE);
 	}
 
-	@Override
-	public BufferAndAvailability getNextBuffer() throws IOException {
+	private AddBacklogMessage getAddBacklogMessage() {
+		int backlog = subpartitionView.getAndResetUnannouncedBacklog();
+		if (backlog > 0) {
+			return new AddBacklogMessage(receiverId, backlog, false);
+		}
+		return null;
```

Comment:
I guess we should not expect `null` return here. As long as the code path enters `getAddBacklogMessage`, then we should guarantee that the respective backlog should be more than 0. 
Maybe add assert backlog instead?

Reply:
It can be null because of redundant data available notification.